spec/requests/restaurants_spec.rb

```diff
@@ -0,0 +1,132 @@
+require 'rails_helper'
+
+RSpec.describe 'Restaurants', type: :request do
+  before(:all) do
+    FactoryBot.create_list(:restaurant, 10)
```

Comment:
I don't like this approach. you create a list of restaurants and `sample` the list. so you can change a record and then use the same record for another test, and that makes the tests a bit coupled. 

I would create in each scope (`describe` or `it`) the object you need **(also read about `let` and `let!`)**, and use the `create_list` only for `get` (also you can randomize the number of restaurants)